### PR TITLE
[STORM-3963] Add m2e plugin to storm-shaded-deps and few minor pom.xml fixes

### PR DESCRIPTION
**Pull request by bipinprasad**
## What is the purpose of the change

*Importing Storm into eclipse and a maven project shows some errors and warnings*

## How was the change tested

*Import into eclipse as well as following github actions*

### CODE REVIEW

File: storm-shaded-deps/pom.xml [hunk 1]
```diff
@@ -358,5 +357,34 @@
                 </executions>
             </plugin>
         </plugins>
+        <pluginManagement>
+            <plugins>
+                <!--This plugin's configuration is used to store Eclipse m2e settings only. It has no influence on the Maven build itself.-->
+                <plugin>
+                    <groupId>org.eclipse.m2e</groupId>
+                    <artifactId>lifecycle-mapping</artifactId>
+                    <version>1.0.0</version>
```
Comment (rzo1) [inline storm-shaded-deps/pom.xml:366]: Shall we pull that version up to easier overview for updating?

**Review by rzo1 — APPROVED** [review]

File: storm-shaded-deps/pom.xml [hunk 2]
```diff
@@ -337,7 +336,7 @@
                 -->
                 <groupId>org.codehaus.mojo</groupId>
                 <artifactId>build-helper-maven-plugin</artifactId>
-                <version>3.3.0</version>
+                <version>3.4.0</version>
```
[the remark this answers is not in the document]
Comment (bipinprasad) [reply storm-shaded-deps/pom.xml:339]: For some reason eclipse is better at understanding the phases on newer one.

File: storm-shaded-deps/pom.xml (hunk 1 above)
Comment (bipinprasad) [reply storm-shaded-deps/pom.xml:366]: Wanted to keep it localized with the goal of removing this section completely when/if eclipse understands the class relocation. But on second thought, will pull this up.

Comment (rzo1) [reply storm-shaded-deps/pom.xml:366]: 🫠

**Pull request merged**